src/commands.rs

```diff
@@ -1003,6 +1027,30 @@ implement_commands! {
         cmd("ZUNIONSTORE").arg(dstkey).arg(keys.len()).arg(keys).arg("AGGREGATE").arg("MAX")
     }
 
+    /// [`Commands::zunionstore`], but with the ability to specify a
+    /// multiplication factor for each sorted set by pairing one with each key
+    /// in a tuple.
+    fn zunionstore_weights<K: ToRedisArgs, W: ToRedisArgs>(dstkey: K, keys: &'a [(K, W)]) {
+        let (keys, weights): (Vec<&K>, Vec<&W>) = keys.iter().map( | (key, weight) | { (key, weight) } ).unzip();
```

Comment:
I get the following error when I do `keys.iter().unzip()`:
```
type mismatch resolving `<std::slice::Iter<'_, (K, W)> as Iterator>::Item == (_, _)`
 expected tuple `(_, _)`
found reference `&(K, W)`rustc[E0271](https://doc.rust-lang.org/error-index.html#E0271)
iterator.rs(3151, 32): required by a bound in `unzip`
```
I _think_ this means unzip requires `Self: Iterator<Item = (_, _)>` but `keys.iter()` iterates over references of the tuple, `&(K, W)` (I'm not 100% sure because hovering over `let i = keys.iter()` with rust-analyzer says that `i: Iter<(K, W)>`, not `i: Iter<&(K, W)>`). 

The mapping returns an iterator that iterates over `(&K, &W)` because the closure allows me to destructure the tuple into individual references to `K` and `W`, which unzip can work with. I wouldn't consider myself super experienced with Rust, so definitely let me know if there's an easier way to do this or I'm misunderstanding the error.

Reply:
Yes, as I said, can you  try with `|item| *item` instead?

Reply:
I'm unable to de-reference any `item` because we can't move it out from behind a shared reference, and `K`, `W`, and `(K, W)` are not copy-able.

Reply:
Ah, okay. Well, in that case just clean up the spacing to look like `map(|(key, weight)| (key, weight))`, please.